### PR TITLE
Hello World in Koka

### LINKED ISSUE #331: Add Hello World in Koka
See https://github.com/koka-lang/koka

### PR DESCRIPTION
**Pull request by BassemMohamed**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [x] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [x] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #331 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Comment by BassemMohamed** [comment]
I am not really sure why, but the older commits ( Hello World in Orc ) is still shown in this pull request, it might need some fixing

Next pull request will be done using a branch to prevent further issues or something.

**Comment by chrboe** [comment]
Regarding the commit mess issue: you're right, you should definitely start a new branch for each PR, this will eliminate lots of problems.

File: archive/k/koka/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Sample Programs in Koka
+
+Welcome to Sample Programs in Koka!
+
+## Sample Programs
+
+- [Hello World in Koka](https://github.com/jrg94/sample-programs/issues/123)
```
[the remark this answers is not in the document]
Comment (jrg94) [reply archive/k/koka/README.md]: Yeah, I noticed this after the fact and fixed it directly.

File: archive/k/koka/hello-world.kk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+function main() 
+{
+  println("Hello, World!")
```
[the remark this answers is not in the document]
Comment (jrg94) [reply archive/k/koka/hello-world.kk]: I would be interested in something like that. I was actually playing with adding a syntax section to the README in Python. I'd like to do that for all the languages but some are better documented than others. 

Also, I'd like to keep the barrier to entry for new languages pretty low. In other words, I'd like to keep the base README format simple, so others can quickly add a new language without investing a ton of time into it. Any additional section like syntax or style would have to be *bonus*. 

Of course, I'm open to whatever. This is a community collection after all.

**Pull request merged**